src/main/java/com/mineinabyss/mobzy/ecs/goals/mobzy/hostile/ThrowItemsGoal.kt

```diff
@@ -117,28 +108,7 @@ class ThrowItemsGoal(
             1.0f,
             1.0f / (Random.nextDouble(0.8, 1.2).toFloat())
         )
-        projectile.shoot(dX, dY, dZ, 1.6f, 12.0f)
-        world.toNMS().addEntity(projectile)
-    }
-}
 
-class DamagingThrownItem(
-    item: ItemStack,
-    val damage: Float,
-    world: World?,
-    thrower: EntityLiving
-) : EntitySnowball(world, thrower) {
-    init {
-        this.item = CraftItemStack.asNMSCopy(item)
-    }
-
-    override fun a(mop: MovingObjectPosition) {
-        super.a(mop)
-
-        if (mop.type == MovingObjectPosition.EnumMovingObjectType.ENTITY) {
-            val hit = (mop as MovingObjectPositionEntity).entity
-            if (hit is EntityPlayer)
-                hit.damageEntity(DamageSource.projectile(this, shooter), damage)
-        }
+        (projectile.toNMS() as EntityProjectileThrowable).shoot(dX, dY, dZ, 1.6f, 12.0f)
```

Comment:
Should be able to use projectile.toNMS<EntityProjectileThrowable>() here

Reply:
`toNMS<EntityProjectileThrowable>()` for the extension functions that are not explicitly defined (you have to make one for each entity type conversion. We could auto generate that actually but :pain:)